### ISSUE TITLE
 Issue with Ray Cluster Connection and Worker Termination

### ISSUE THREAD

**Issue by sip-aravind-g** [ray-project/ray#42350]
### What happened + What you expected to happen

Preface:
I am currently facing an issue with my Ray cluster setup and worker termination. The relevant log entries are provided below for your reference:

ERROR:
Took 0.010474920272827148 seconds to parse the hackernews submissions
Accuracy on the training set is 1.0
Accuracy on the test set is 0.5
2024-01-11 11:59:36,206 INFO worker.py:1329 -- Using address 10.224.149.138:6379 set in the environment variable RAY_ADDRESS
2024-01-11 11:59:36,207 INFO worker.py:1458 -- Connecting to an existing Ray cluster at address: 10.224.149.138:6379...
[2024-01-11 11:59:36,297 I 2245 2245] global_state_accessor.cc:368: This node has an IP address of 10.224.226.71, but we cannot find a local Raylet with the same address. This can happen when you connect to the Ray cluster with a different IP address or when connecting to a container.
2024-01-11 11:59:36,367 INFO worker.py:1633 -- Connected to Ray cluster. View the dashboard at http://10.224.149.138:8265
[2024-01-11 12:00:05,393 E 2245 2306] core_worker_process.cc:216: Failed to get the system config from raylet because it is dead. Worker will terminate. Status: GrpcUnavailable: RPC Error message: failed to connect to all addresses; last error: UNKNOWN: ipv4:10.224.226.71:37287: tcp handshaker shutdown; RPC Error details: .Please see `raylet.out` for more details.

Description:
The issue arises during the connection to the Ray cluster, and it seems to be related to the IP address mismatch or connectivity problems with the Raylet. The worker terminates with an error indicating a failure to retrieve the system config from the Raylet, and the status is reported as GrpcUnavailable.

Steps Taken:
I have verified the IP addresses and connection parameters, and they appear to be correct. However, the worker termination issue persists. I also checked the Ray dashboard for additional insights.

Next Steps:

Please advise on troubleshooting steps to identify and resolve the connectivity issue with the Raylet.
Any recommendations on further logs or debugging information that could provide more insights into the problem.
Suggestions on possible misconfiguration or additional settings that may be contributing to the worker termination.
I appreciate your assistance in resolving this matter.

### Versions / Dependencies

ray version 2.7.0

### Reproduction script

[hyperparamter.zip](https://github.com/ray-project/ray/files/13908321/hyperparamter.zip)

### Issue Severity

High: It blocks me from completing my task.

**Comment by sip-aravind-g** [comment]
@jjyao Please take a look when you can. Thanks!

**Comment by sip-aravind-g** [comment]
This issue I am investigated intermittently appearing, sometimes YES, sometimes NO. 
if we enable entrypoint_num_cpus=1, then the ray cluster spanning the worker-node and the job executing successfully.

**Comment by rkooo567** [comment]
1. Is 10.224.226.71 the address of worker or head?
2. When this happens, if you use ray list nodes in a head node, can you find the IP 10.224.226.71?

**Comment by sip-aravind-g** [comment]
@rkooo567 that’s a head node

**Comment by jjyao** [comment]
Are you using kuberay?

**Comment by sip-aravind-g** [comment]
Yes @jjyao

**Comment by pxp531** [comment]
We also encountered the same problem and had no problems with the initial submission. At the same time, we configured the autoscaler policy, which will delete the pod in 120 seconds. If you create a new job at this time, it will pull up a new pod, but the original pod is not there, theoretically the ip of the pod is not there, it looks like it has been walking the cache, and it is not very familiar with this principle, so it is just some speculation. We also use kuberay

**Comment by pxp531** [comment]
> 这个问题我被调查断断续续地出现，有时是，有时不是。如果我们启用 entrypoint_num_cpus=1，则跨 Worker 节点的 Ray 集群和作业将成功执行。

Where is this parameter set? Can all tasks be set this way

**Comment by pxp531** [comment]
> entrypoint_num_cpus

I set this parameter according to your method and then there is no problem to submit, kuberay will pull up a new pod; But the principles and problems of this have not been fully understood
